module/blobs/blob_io.go

```diff
@@ -64,7 +64,9 @@ func (bw *BlobChannelWriter) Write(data []byte) (int, error) {
 // sendNewBlob sends the currently buffered data to the blob channel and resets the buffer.
 func (bw *BlobChannelWriter) sendNewBlob() {
 	blob := NewBlob(bw.buf.Bytes())
-	bw.blobs <- blob
+	if bw.blobs != nil {
```

Comment:
why would `bw.blobs` be nil here?

Reply:
its initialized by parameter in a constructor